enhancements/cli-in-cluster-management.md

```diff
@@ -0,0 +1,164 @@
+    * With krew, artifacts will be packaged within an image.  Each cluster will have an index of CLI images.  Mirrored local registries will provide binaries to disconnected environments.  
+
+Option 4 is the option we as a team are leaning toward.  It requires that each OpenShift CLI publish artifacts, for example, [index of /pub/openshift-v4/clients](https://mirror.openshift.com/pub/openshift-v4/clients/)
+then create an image similar to this: [example cli-artifacts image](https://github.com/openshift/oc/blob/master/images/cli-artifacts/Dockerfile.rhel).  CLI artifacts are packaged in images.
+The CLI in-cluster manager will gather information from each CLI's CR about each CLI and its image.  A Krew index will be populated in every cluster to provide available CLIs and plugins for a cluster.
+
+### Non-Goals
+
+* CLI manager will not build or serve the binaries.  It will know where to find them.
+* This proposal is not concerned with _which_ binaries will be managed.  This proposal is meant to determine the mechanism only.  Consumers and publishers are clients of the mechanism. 
+* CLI manager will not create or update the ClusterCLI Custom Resources, those will be managed by individual CLIs and their operators. 
+
+### Requirements:
+
+1. CLI author requirements:
+    - provide CLI image containing binaries
+    - Custom Resource that describes the binaries (location within the image, metadata, description)
+2. An operator that will be responsible for:
+    - registering the new CLI CRD 
+    - reading metadata from CR from authors and extracting the binaries from the CLI images to local disk
+    - serving extracted binaries
+    - serving krew-compatible index
+3. oc changes:
+    - embed krew
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a user, I want a CLI in-cluster manager for various CLIs available for OpenShift and related services so that I can discover, install and list them
+The user will invoke the following commands:
+
+* `oc krew list` to view installed plugins
+* `oc krew search` will communicate with the krew index served by the cli-in-cluster-manager operator and will retrieve information about CLIs
+* `oc krew install foo` will install `foo` to a user's system.
+
+##### example:
+```
+$ oc krew list
+NAME                 DESC                        LATEST       INSTALLED
+-----                -----                       -----        -----
+openshift-installer  OpenShift Installer CLI     4.4          4.3
+kubectl              Kubernetes CLI              1.15         1.13
+oc                   OpenShift CLI               4.4          4.3
+odo                  OpenShift Developer CLI     1.0          Not Installed
+kn                   Knative CLI                 1.0          Not Installed
+tkn                  Tekton CLI                  1.0          Not Installed
+
+```
+
+## Design Detals
```

Comment:
I expect (as noted on slack) a discussion of version compatibility between cluster versions.  Also, need to describe which metadata from OLM operators applies to you (since they have similar problems, compatibility, etc).